src/middlewares/createInsightsMiddleware.ts

```diff
@@ -126,7 +127,14 @@ export const createInsightsMiddleware: CreateInsightsMiddleware = props => {
           if (onEvent) {
             onEvent(event, _insightsClient);
           } else if (event.insightsMethod) {
-            insightsClient(event.insightsMethod, event.payload);
+            if (hasUserToken()) {
+              insightsClient(event.insightsMethod, event.payload);
+            } else {
+              warning(
+                false,
+                'Cannot send event to Algolia Insights because `userToken` is not set.'
+              );
```

Comment:
Why don't we test that this warning is correctly displayed?

Reply:
added https://github.com/algolia/instantsearch.js/pull/4724/commits/8c489dd5aebe325d3d10a88b974b1b5f53db52d0